### PR TITLE
farrah/fix navbar language switcher re-render

### PR DESCRIPTION
**Pull request by farrah-deriv**
Changes:

-   Fixed navbar language switcher re-render

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [x] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/components/hooks/use-client-information.js [hunk 1]
```diff
@@ -1,26 +1,28 @@
...
+    const setCurrentClientInformation = () => {
+        const current_client_information = getClientInformation(getDomain())
+        if (JSON.stringify(current_client_information) != JSON.stringify(client_information)) {
+            setClientInformation(current_client_information)
+        }
+    }
+
+    useEffect(() => {
+        callback_ref.current = setCurrentClientInformation
+    })
```
Comment (yashim-deriv) [inline src/components/hooks/use-client-information.js:17]: This will rerun the operations inside `useEffect` on every re-render.  Is this intended?

Comment (farrah-deriv) [reply src/components/hooks/use-client-information.js:17]: Yes, this is to save the latest callback so we can use the updated state in the interval.

**Comment by sean-binary** [comment]
I'm getting a white screen on the test link 🙏

File: src/components/hooks/use-client-information.js [hunk 2]
```diff
@@ -1,26 +1,31 @@
...
 
 export const useClientInformation = () => {
     const [client_information, setClientInformation] = useState(false)
-    const [current_client_information, setCurrentClientInformation] = useState(false)
+    const callback_ref = useRef()
+
+    const setCurrentClientInformation = () => {
+        const current_client_information = getClientInformation(getDomain())
+        if (
+            current_client_information &&
```
Comment (habib-deriv) [inline src/components/hooks/use-client-information.js:11]: Lets put this logic inside a separate variable and use inside the if statement

**Review by habib-deriv — APPROVED** [review]
LGTM

File: src/components/hooks/use-client-information.js (hunk 1 above)
Comment (sean-binary) [reply src/components/hooks/use-client-information.js:19]: It is quite a bad practice to go without a dependency array. Maybe as a suggestion, we could use a helper function like this instead

```
const useCallbackRef = (callback) => {
    const callbackRef = useRef(callback)

    useLayoutEffect(() => {
        callbackRef.current = callback
    }, [callback])

    return callbackRef
}
```

**Comment by suthesan-deriv** [comment]
Please help to resolve the conflicts @farrah-deriv 🙏

**Review by yashim-deriv — APPROVED** [review]
LGTM

**Review by habib-deriv — APPROVED** [review]

**Pull request merged**